src/vault/client/http.clj

```diff
@@ -122,40 +123,44 @@
 
 (defn- authenticate-token!
   "Updates the token ref by storing the given auth token."
-  [auth-ref token]
+  [client token]
   (when-not (string? token)
     (throw (IllegalArgumentException. "Token credential must be a string")))
-  (reset! auth-ref {:client-token (str/trim token)}))
+  (reset! (:auth client) {:client-token (str/trim token)}))
 
 
 (defn- authenticate-userpass!
   "Updates the token ref by making a request to authenticate with a username
   and password."
-  [auth-ref api-url credentials]
+  [client credentials]
```

Comment:
These are private functions, so the contract should be pretty clear by code proximity.